src/app/zone-egresses/data/index.ts

```diff
@@ -0,0 +1,93 @@
+import type { PaginatedApiListResponse as CollectionResponse } from '@/types/api.d'
+import type {
+  ZoneEgressOverview as PartialZoneEgressOverview,
+  ZoneEgress as PartialZoneEgress,
+  KDSSubscription,
+} from '@/types/index.d'
+
+type PartialZoneEgressInsight = any
+type PartialInternalZoneEgress = PartialZoneEgressOverview['zoneEgress']
+
+// TODO(jc) Theres probably a better way to not copy/pasta this i.e. make `Entity` composable
+type InternalZoneEgress = {
+  socketAddress: string
+} & PartialInternalZoneEgress
+
+export type ZoneEgress = {
+  config: PartialZoneEgress
+  socketAddress: string
+} & PartialZoneEgress
+// end TODO
+
+// TODO(jc) currently in our manually written types ZoneEgressInsight is any therefore
+// we have no Partial*
+export type ZoneEgressInsight = {
+  connectedSubscription?: KDSSubscription
+  subscriptions: KDSSubscription[]
+}
+export type ZoneEgressOverview = PartialZoneEgressOverview & {
+  zoneEgress: InternalZoneEgress
+  zoneEgressInsight?: ZoneEgressInsight
+  state: 'online' | 'offline'
+}
+// TODO(jc) Theres probably a better way to not copy/pasta this i.e. make `Entity` composable
+const InternalZoneEgress = {
+  fromObject: (item: PartialInternalZoneEgress): InternalZoneEgress => {
+    return {
+      ...item,
+      socketAddress: item.networking?.address && item.networking?.port ? `${item.networking.address}:${item.networking.port}` : '',
+    }
+  },
+}
+
+export const ZoneEgress = {
+  fromObject: (item: PartialZoneEgress): ZoneEgress => {
+    return {
+      ...item,
+      config: item,
+      socketAddress: item.networking?.address && item.networking?.port ? `${item.networking.address}:${item.networking.port}` : '',
+    }
+  },
+}
+// end TODO
+export const ZoneEgressInsight = {
+  fromObject: (item?: PartialZoneEgressInsight): ZoneEgressInsight | undefined => {
+    // if item isn't set don't even try augmenting things
+    return isSet<PartialZoneEgressInsight>(item)
+      ? ((item) => {
+        const subscriptions: KDSSubscription[] = Array.isArray(item.subscriptions) ? item.subscriptions : []
```

Comment:
```suggestion
        const subscriptions: XDSSubscription[] = Array.isArray(item.subscriptions) ? item.subscriptions : []
```

Reply:
I'm guessing this might be that there is an XDSSubscription type in the backend somewhere, or maybe we just have the wrong name for the type? FYI: This was a KDSSubscription before this PR. If we do need to add additional type to model an XDSSubscription or change the name I think we should do that after this PR. I'm gonna resolve this for now, but feel free to unresolve if you think we should address this here. It would also be good to understand why we don't already have an XDSSubscription type in the frontend.